From 87c0bd44a43769905f3bb9bea4c8af307c58a79b Mon Sep 17 00:00:00 2001
From: Xiang Zheng <zhengxiang9@huawei.com>
Date: Fri, 17 Apr 2020 15:05:54 +0800
Subject: [PATCH 1/2] CryptoPkg/OpensslLib: Modify process_files.pl for OpenSSL
 1.1.1f

Before upgrading the version of OpenSSL to 1.1.1f, we need to update
process_files.pl so that we can auto-generate the correct files.

This patch may confict with the opensource codes in the future, than
let it go with opensouce.

Change-Id: Id17969bab444bad8cdd139258ef53d5eed91ff88
Signed-off-by: Xiang Zheng <zhengxiang9@huawei.com>
---
 .../Library/Include/{internal => crypto}/dso_conf.h    |  0
 CryptoPkg/Library/OpensslLib/process_files.pl          | 10 +++++-----
 2 files changed, 5 insertions(+), 5 deletions(-)
 rename CryptoPkg/Library/Include/{internal => crypto}/dso_conf.h (100%)

diff --git a/CryptoPkg/Library/Include/internal/dso_conf.h b/CryptoPkg/Library/Include/crypto/dso_conf.h
similarity index 100%
rename from CryptoPkg/Library/Include/internal/dso_conf.h
rename to CryptoPkg/Library/Include/crypto/dso_conf.h
diff --git a/CryptoPkg/Library/OpensslLib/process_files.pl b/CryptoPkg/Library/OpensslLib/process_files.pl
index bbcfa0d0..a99ad866 100755
--- a/CryptoPkg/Library/OpensslLib/process_files.pl
+++ b/CryptoPkg/Library/OpensslLib/process_files.pl
@@ -109,8 +109,8 @@ BEGIN {
             # Generate dso_conf.h per config data
             system(
                 "perl -I. -Mconfigdata util/dofile.pl " .
-                "crypto/include/internal/dso_conf.h.in " .
-                "> include/internal/dso_conf.h"
+                "include/crypto/dso_conf.h.in " .
+                "> include/crypto/dso_conf.h"
                 ) == 0 ||
                     die "Failed to generate dso_conf.h!\n";
 
@@ -264,9 +264,9 @@ copy($OPENSSL_PATH . "/include/openssl/opensslconf.h",
      $OPENSSL_PATH . "/../../Include/openssl/") ||
    die "Cannot copy opensslconf.h!";
 print "Done!";
-print "\n--> Duplicating dso_conf.h into Include/internal ... ";
-copy($OPENSSL_PATH . "/include/internal/dso_conf.h",
-     $OPENSSL_PATH . "/../../Include/internal/") ||
+print "\n--> Duplicating dso_conf.h into Include/crypto ... ";
+copy($OPENSSL_PATH . "/include/crypto/dso_conf.h",
+     $OPENSSL_PATH . "/../../Include/crypto/") ||
    die "Cannot copy dso_conf.h!";
 print "Done!\n";
 
-- 
2.19.1

